R/file_utils.R

```diff
@@ -89,9 +89,11 @@ get_pjn_data <- function(pjnz_path) {
 #' specio:::get_pjnz_csv_data(pjnz, "PJN")
 get_pjnz_csv_data <- function(pjnz_path, extension) {
   file <- get_filename_from_extension(extension, pjnz_path)
-  ## Use UTF-8-BOM encoding to remove the Byte order mark at the start of the
-  ## file if it is present. This is ignored by default on Linux but included
-  ## when reading the file on windows.
-  csv <- utils::read.csv(unz(pjnz_path, file), stringsAsFactors = FALSE,
-                         encoding = "UTF-8-BOM")
+  csv <- utils::read.csv(unz(pjnz_path, file), stringsAsFactors = FALSE)
+  ## Set column name of first column manually to remove any possible byte order
+  ## mark added when reading the file on windows. We should be able to work
+  ## around this by passing encoding = "UTF-8-BOM" or fileEncoding = "UTF-8-BOM"
+  ## as arg to read.csv but this isn't working reliably.
+  colnames(csv)[[1]] <- "Tag"
```

Comment:
How about testing only for the bad case here rather than doing this unconditionally? Is it possible to add a test with a file that shows this behaviour too, or is this dependent on the machine that it is run on?

Reply:
if you pass `check.names = FALSE` to `read.csv` we might get a reasonably good shot at identifying the bad cases directly.

We can set up appveyor testing too, and possibly also mock `read.csv` here on linux?

Reply:
```
length(b) > 3 && identical(tail(b, 3), charToRaw("Tag"))
```